packages/core/src/connectors/index.ts

```diff
@@ -33,6 +33,42 @@ export const getConnectorInstanceById = async (id: string): Promise<ConnectorIns
   return { connector, ...found };
 };
 
+const isSmsConnector = (connector: ConnectorInstance): connector is SmsConnector => {
+  return connector.metadata.type === ConnectorType.SMS;
+};
+
+export const getSmsConnectorInstance = async (): Promise<SmsConnector> => {
+  const connectors = await getConnectorInstances();
+  const smsConnector = connectors
+    .filter((connector) => connector.connector?.enabled)
+    .find((connector) => isSmsConnector(connector));
```

Comment:
Maybe specify generic to avoid call isSmsConnector twice?

Reply:
wondering can we merge as one method getConnectorInstance<T extends ConnectorInstance > = async ( type: connectorType ): Promise(T);

Reply:
refactored